src/gromacs/nblib/topology.h

```diff
@@ -70,6 +70,24 @@ std::vector<gmx::ExclusionBlock> toGmxExclusionBlock(const std::vector<std::tupl
 // Add offset to all indices in inBlock
 std::vector<gmx::ExclusionBlock> offsetGmxBlock(std::vector<gmx::ExclusionBlock> inBlock, int offset);
 
+//! Helper class for Topology to keep track of particle IDs
+class EnumerationKey
+{
+    // store by (molecule name, molecule nr, residue name, particle name)
+    using DataType =
+            std::unordered_map<std::string, std::unordered_map<int, std::unordered_map<ResidueName, std::unordered_map<ParticleName, int>>>>;
+
+public:
+    //! build sequence from a list of molecules
+    void enumerate(const std::vector<std::tuple<Molecule, int>>&);
```

Comment:
In particular, this is not such a good name, because enumerate would seem to imply that you will get an enumeration back, as an iterator. I would call this something like build().

Reply:
yes, build() would be much better